src/scripts/main.js

```diff
@@ -1,3 +1,21 @@
 'use strict';
 
-// write code here
+const allLi = document.querySelectorAll('li');
+const tree = document.querySelector('.tree');
+
+for (const li of allLi) {
+  if (li.children.length !== 0) {
+    const span = document.createElement('span');
+
+    li.prepend(span);
+    span.append(span.nextSibling);
+  }
+}
+
+tree.addEventListener('click', function (e) {
+  if (e.target.tagName !== 'SPAN') {
+    return;
+  }
+
+  e.target.nextSibling.hidden = !e.target.nextSibling.hidden;
+});
```

Comment:
The click event listener on the tree toggles the visibility of the next sibling of the clicked `span`. However, this assumes that the next sibling is always the `ul` that should be shown or hidden. If there are other elements between the `span` and the `ul`, or if the `ul` is not the next sibling for some other reason, this will not work as expected. It would be more reliable to specifically select the `ul` that should be shown or hidden. One way to do this would be to use `e.target.parentElement.querySelector('ul')` instead of `e.target.nextSibling`.